### ISSUE TITLE
Unable to communicate with Duplicati

### ISSUE THREAD

**Issue by etrigan63** [linuxserver/Heimdall-Apps#634]
I have tried various methods to connect the Heimdall Duplicati app to show details from my Duplicati container on TureNAS Scale. The error presented is either 

` Error connecting to Duplicati: Error getting nonce `

I get the same error using the internal DNS name.

**Comment by mvdkleijn** [comment]
If you're getting that error, the Duplicati app in Heimdall is attempting to connect to `<your-url>/login.cgi` to retrieve the nonce.

It is either not getting a response (i.e. cannot reach the Duplicati server) or Duplicati is returning some sort of error.

Please ensure you can reach the above mentioned URL manually. Also check your Heimdall / Duplicati logs to see if they output anything.

**Comment by mvdkleijn** [comment]
Closing this to clean up, feel free to reopen or comment when needed.

Reasoning:
- no further information received

**Comment by pbru87** [comment]
I also tried to get this to work, but unfortunately failed.
I tried using an easier password in Duplicati, tried to change some Duplicati settings, but nothing worked.

Both Heimdall and Duplicati work just fine. I was even able to use Heimdall's "Enhanced Apps" feature successfully with other apps. There the API feature works just fine.

I'm curious, is that Duplicati API call working for any Heimdall user?

My setup:
- Windows 11
- Heimdall running from local Docker Desktop App.
- Local install of Duplicati.

Would be great to receive some tips. Thanks in advance!

**Comment by j0nnymoe** [comment]
Use your host IP - localhost will reference inside the container.

**Comment by pbru87** [comment]
@j0nnymoe: Thanks for the tip! That gave me some hope to look into it again.

It was a bit tricky, but I was able to solve it now.

**Solution for Windows (but should work similar with some other OS, I guess):**
1. When the Docker container is running in network mode "bridged", its host is available as "**host.docker.internal**" (that seems to be Windows only - for Linux, it seems to be "**172.17.0.1**").
2. But you cannot just enter "**host.docker.internal**" in Heimdall's url field.
3. Instead, go to your container in Docker Destkop App, open the console and do `$ ping host.docker.internal`. Now, you will see the IP address it resolves to, the IP of its host. In my case this was "192.168.65.254".
4. I then entered "**http://192.168.65.254:8200/**" into Heimdall's URL field for Duplicati - and it worked.

I hope this helps anyone else in the future.

@etrigan63: Maybe also a solution for you? (In case it still bugs you.)
